state/autocertcache_test.go

```diff
@@ -0,0 +1,44 @@
+package state_test
+
+import (
+	jc "github.com/juju/testing/checkers"
+	"golang.org/x/crypto/acme/autocert"
+	"golang.org/x/net/context"
+	gc "gopkg.in/check.v1"
+
+	statetesting "github.com/juju/juju/state/testing"
+)
+
+type autocertCacheSuite struct {
+	statetesting.StateSuite
+}
+
+var _ = gc.Suite(&autocertCacheSuite{})
+
+func (s *autocertCacheSuite) TestCache(c *gc.C) {
+	ctx := context.Background()
+	cache := s.State.AutocertCache()
+	err := cache.Put(ctx, "a", []byte("aval"))
+	c.Assert(err, jc.ErrorIsNil)
+	err = cache.Put(ctx, "b", []byte("bval"))
+	c.Assert(err, jc.ErrorIsNil)
+
+	data, err := cache.Get(ctx, "a")
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(string(data), gc.Equals, "aval")
+
+	data, err = cache.Get(ctx, "b")
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(string(data), gc.Equals, "bval")
+
+	data, err = cache.Get(ctx, "c")
+	c.Assert(err, gc.Equals, autocert.ErrCacheMiss)
+	c.Assert(data, gc.IsNil)
+
+	err = cache.Delete(ctx, "b")
```

Comment:
Try to delete another time? I kind of feel that these could be different tests, like one for a notFound, one for fetting the cache, one for overriding a name, one for deleting, even twice. Take it or leave it.